### PR TITLE
Modify ConfigValueLocation to use URL instead of String

### PR DESCRIPTION
**Pull request by jcazevedo**
As suggested by @derekmorr in #178, this PR modifies `ConfigValueLocation` to use `Path` instead of `String` to encode the location of failures.

### CODE REVIEW

**Comment by derekmorr** [comment]
This looks good to me, but @leifwickland raised some concerns in https://github.com/melrief/pureconfig/pull/178#issuecomment-288207971

**Comment by jcazevedo** [comment]
I've modified `ConfigValueLocation` to use `URL` instead of `Path`, in order to be able to encode the location of configuration files inside `.jar` files.

**Comment by derekmorr** [comment]
`Path` should be able to handle files in a jar file, but arguable using `URL` is better since `ConfigFactory` can load files from` URL`s.

**Comment by melrief** [comment]
I prefer `URL` because a `URL` can identify any resource in the system and can be used for locations that are not necessarily files.

**Review by melrief — APPROVED** [review]
Looks good to me

**Comment by jcazevedo** [comment]
Since `ConfigOrigin` has `url` defined in cases where `filename` is `null`, I also think we're better off with `URL`.

**Comment by melrief** [comment]
I approved this, let's wait for another approval and merge.

**Comment by derekmorr** [comment]
Approved.

**Pull request merged**
